src/Vector3d.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_SG_VECTOR3D_H_
+#define LSST_SG_VECTOR3D_H_
+
+/// \file
+/// \author Serge Monkewitz
+/// \brief This file declares a class for representing vectors in ℝ³.
+
+#include <cmath>
+#include <iosfwd>
+#include <limits>
+#include <stdexcept>
+
+
+namespace lsst {
+namespace sg {
+
+// Forward declarations
+class Angle;
+class UnitVector3d;
+
+
+/// `Vector3d` is a vector in ℝ³ with components stored in double precision.
+class Vector3d {
+public:
+    /// The default constructor creates a zero vector.
+    Vector3d() { _v[0] = 0.0; _v[1] = 0.0; _v[2] = 0.0; }
+    /// This constructor creates a vector with the given components.
+    Vector3d(double x, double y, double z) { _v[0] = x; _v[1] = y; _v[2] = z; }
+
+    bool operator==(Vector3d const & v) const {
+        return _v[0] == v._v[0] && _v[1] == v._v[1] && _v[2] == v._v[2];
+    }
+    bool operator!=(Vector3d const & v) const {
+        return _v[0] != v._v[0] || _v[1] != v._v[1] || _v[2] != v._v[2];
+    }
+    bool operator<(Vector3d const & v) const {
+        if (_v[0] == v._v[0]) {
+            if (_v[1] == v._v[1]) {
+                return _v[2] < v._v[2];
+            }
+            return _v[1] < v._v[1];
+        }
+        return _v[0] < v._v[0];
+    }
```

Comment:
Should go in .cc file - if you even need it.  For objects that don't have an obvious sort order, I tend to lean towards not providing a sorting operator on the class, and passing a comparison functor as a template argument if you need it for a particular container.